src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,174 @@
+        int month3;
+        month3 = Integer.parseInt(dateFrom.substring(sixthIndex + 5, sixthIndex + 6));
+        if (month3 == 0) {
+            month5 = Integer.parseInt(dateFrom.substring(sixthIndex + 6,
+                    sixthIndex + 7));//monthFrom
+        } else {
+            month5 = Integer.parseInt(dateFrom.substring(sixthIndex + 5,
+                    sixthIndex + 7));//monthFrom
+        }
+
+        //here I find data to output
+        int day4;
+        int year3;
+        year3 = Integer.parseInt(dateFrom.substring(sixthIndex + 8, sixthIndex + 12)); //yearFrom
+        day4 = Integer.parseInt(dateTo.substring(ninthIndex, ninthIndex + 1));
+
+        //the same about day from String 'dateTo'
+        if (day4 == 0) {
+            day6 = Integer.parseInt(dateTo.substring(ninthIndex + 1, ninthIndex + 2));//dayTo
+        } else {
+            day6 = Integer.parseInt(dateTo.substring(ninthIndex, ninthIndex + 2));//dayTo
+        }
+
+        //the same about month from String 'dateTo'
+        int month4;
+        month4 = Integer.parseInt(dateTo.substring(ninthIndex + 3, ninthIndex + 4));
+        if (month4 == 0) {
+            month6 = Integer.parseInt(dateTo.substring(ninthIndex + 4, ninthIndex + 5));//monthTo
+        } else {
+            month6 = Integer.parseInt(dateTo.substring(ninthIndex + 3, ninthIndex + 5)); //monthTo
+        }
+        //the same about year from String 'dateTo'
+        int year4;
+        StringBuilder result = new StringBuilder();
+        year4 = Integer.parseInt(dateTo.substring(ninthIndex + 6, ninthIndex + 10)); //yearTo
+        result.append("Report for period " + data1 + " - " + data2);
+
+        //here we find day and month from array 'date'
+        int firstIndex = 0;
+        for (int i = 0; i < data.length; i++) {
+            day1 = Integer.parseInt(data[i].substring(0, 1));
+            month1 = Integer.parseInt(data[i].substring(tenthIndex + 1,
+                    tenthIndex + 2));
+
+            //here is the same logic as we find everything before, but we find this in array 'date'
+            if (day1 == 0) {
+                day = Integer.parseInt(data[i].substring(1, 2)); //day in data
+            } else {
+                day = Integer.parseInt(data[i].substring(0,
+                        tenthIndex)); //day in data
+            }
+            if (month1 == 0) {
+                month = Integer.parseInt(data[i].substring(tenthIndex + 1,
+                        eleventhIndex)); //month in data
+            } else {
+                month = Integer.parseInt(data[i].substring(tenthIndex,
+                        eleventhIndex)); //month in data
+            }
+            year = Integer.parseInt(data[i].substring(eleventhIndex + 1,
+                    twelfthIndex)); // year in data
+
+            //we check if data from array 'data' is in limits of 'dateFrom' and 'dateTo'
+            if (day >= day5 && day <= day6 && month >= month5 && month <= month6
```

Comment:
Don't create redundant loops. The nested loop on line 164 is unnecessary; you can combine the logic with the loop starting on line 113.